routing/router.go

```diff
@@ -1051,76 +1122,7 @@ func (r *ChannelRouter) networkHandler() {
 			validationBarrier.InitJobDependencies(update.msg)
 
 			r.wg.Add(1)
-			go func() {
```

Comment:
I think the loop scope bug only happens if there is a for loop iterating over something: https://github.com/golang/go/discussions/56010

See also this toy example that mimics this code and doesn't redefine the variable: https://go.dev/play/p/wKd82-WVsR-

vs this example that uses a for loop and does redefine the variable: https://go.dev/play/p/BfBCi5k2LNw

Reply:
yeah you are right. I mistakenly thought it was the from receiving from the chan. Updated the description plus the commit message.